src/app/js/formats/streamgraph/Streamgraph.js

```diff
@@ -0,0 +1,476 @@
+import React, { PureComponent } from "react";
+import compose from "recompose/compose";
+import { StyleSheet, css } from "aphrodite/no-important";
+import * as d3 from "d3";
+import {
+  zoomFunction,
+  distinctColors,
+  transformDataIntoMapArray,
+  getMinMaxValue
+} from "./utils";
+import injectData from "../injectData";
+import exportableToPng from "../exportableToPng";
+
+const styles = StyleSheet.create({
+  divContainer: {
+    overflow: "hidden",
+    position: "relative"
+  },
+  tooltip: {
+    position: "absolute"
+  },
+  vertical: {
+    marginTop: 60,
+    height: 190,
+    pointerEvents: "none"
+  },
+  legend: {
+    position: "relative"
+  },
+  legendItem: {
+    marginTop: 2
+  },
+  legendItemText: {
+    marginLeft: 5
+  },
+  legendButton: {
+    height: 15,
+    width: 15
+  }
+});
+
+class Streamgraph extends PureComponent {
+  constructor(props) {
+    super(props);
+    this.state = {
+      width: 800,
+      height: 300
+    };
+
+    this.divContainer = React.createRef();
+    this.svgContainer = React.createRef();
+    this.anchor = React.createRef();
+
+    this.xAxisScale;
+    this.yAxisScale;
+    this.xAxis;
+    this.yAxisL;
+    this.yAxisR;
+    this.gx;
+    this.gyr;
+    this.gyl;
+    this.streams;
+
+    this.hoveredKey;
+    this.hoveredValue;
+    this.mouseIsOverStream = false;
+
+    this.zoomFunction = zoomFunction.bind(this);
+  }
+
+  initTheGraphBasicsElements(width, height, margin, svgViewport) {
+    const zoom = d3
+      .zoom()
+      .scaleExtent([1, 32])
+      .translateExtent([[0, 0], [width, height]])
+      .extent([[0, 0], [width, height]])
+      .on("zoom", this.zoomFunction);
+
+    const innerSpace = svgViewport
+      .append("g")
+      .attr("width", width)
+      .attr("class", "inner_space")
+      .attr("transform", "translate(" + margin.left + "," + margin.top + ")");
+
+    innerSpace
+      .append("defs")
+      .append("clipPath")
+      .attr("id", "mask")
+      .append("rect")
+      .attr("width", width)
+      .attr("height", height);
+
+    const graphZone = innerSpace
+      .append("g")
+      .attr("width", width)
+      .attr("height", height)
+      .attr("clip-path", "url(#mask)")
+      .call(zoom);
+
+    graphZone
+      .append("rect")
+      .attr("width", "100%")
+      .attr("height", "100%")
+      .attr("opacity", 0);
+
+    return [innerSpace, graphZone];
```

Comment:
It's very pythonic!
But prefer to return an object, in this way you can use object destructuring later.

```js
return { innerSpace, graphZone };
```